File: lab_2/part1/prng.cpp
#include <iostream>
#include <random>
#include <ctime>

#define MAX_BIT 128

using namespace std;

int generator()
{
    srand(time(0));
    for (int i = 0; i < MAX_BIT; ++i)
    {
        unsigned long long rand_num = rand() % 32767;
        bool binary_num = rand_num % 2;
        cout << binary_num;
    }
    return 0;
}

int main()
{
    cout << generator() << endl;
}

File: lab_2/part1/prng.java
import java.util.Random;

public class RandomBitSequence {

    private static final int MAXBIT = 128;
/**
     * Генерирует случайную битовую последовательность заданной длины.
     *
     * @param length Длина генерируемой последовательности (должна быть меньше или равна MAXBIT).
     * @return Случайная битовая последовательность в виде строки.
     */
    public static void main() {

        Random rand = new Random(System.currentTimeMillis());

        for (int i = 0; i < MAXBIT; i++) {
            long randNum = rand.nextInt(32767);
            boolean binaryNum = randNum % 2 == 1;
            System.out.print(binaryNum ? "1" : "0");
        }
    }
}